fault.c

```diff
@@ -38,46 +85,56 @@ static struct kibosh_fault_unreadable *kibosh_fault_unreadable_parse(json_value
     struct kibosh_fault_unreadable *fault = NULL;
     json_value *code_obj = NULL;
     json_value *prefix_obj = NULL;
+    json_value *file_type_obj = NULL;
 
     code_obj = get_child(obj, "code");
     if ((!code_obj) || (code_obj->type != json_integer)) {
-        INFO("kibosh_fault_unreadable_parse: No valid \"code\" field found in fault object.\n");
+        INFO("%s: No valid \"code\" field found in fault object.\n", __func__);
         goto error;
     }
     prefix_obj = get_child(obj, "prefix");
     if ((!prefix_obj) || (prefix_obj->type != json_string)) {
-        INFO("kibosh_fault_unreadable_parse: No valid \"prefix\" field found in fault object.\n");
+        INFO("%s: No valid \"prefix\" field found in fault object.\n", __func__);
+        goto error;
+    }
+    file_type_obj = get_child(obj, "file_type");
+    if ((!file_type_obj) || (file_type_obj->type != json_string)) {
```

Comment:
Originally, I made file_type optional, however, since Kibosh uses Json strings to specify faults, it seems more robust to make each Kibosh Json string having fixed fields. If user set file_type to "", it still means that Kibosh will apply fault to all the files. In the python wrapper for Ducktape, the type_file field is optional with default values, so it is still easy to use for the users.